### PR TITLE
autoupdater: fix version compare bugs

### LINKED ISSUE #208: autoupdater: Firmware version 1.7.0 not considered newer than 1.7.0~rc1
It used to be the case that (following Debian's handling of version numbers) firmware version 1.7.0 was considered newer than 1.7.0\~rc1.  However, with the latest gluon 2018.1.3, that does not seem to work any more: We have rolled out 1.7.0\~rc1 on our beta branch, and now we built 1.7.0, but even when doing a force update check the autoupdater just says "No new firmware available."

### PR DESCRIPTION
**Pull request by NeoRaider**
Fixes two bugs in the autoupdater (one regression, one long-standing issue).

Closes #208

### CODE REVIEW

File: admin/autoupdater/src/version.c [hunk 1]
```diff
@@ -34,8 +34,10 @@ static int char_order(char c) {
 		return 0;
 	else if (isalpha(c))
 		return c;
-	else if (c == '~')
+	else if (c == 0)
```
Comment (jplitza) [inline admin/autoupdater/src/version.c]: Everywhere else we use `'\0'` instead of `0` for comparison.
```suggestion
	else if (c == '\0')
```

**Comment by NeoRaider** [comment]
@jplitza At first, I wanted to make a change similar to your fixup, but dpkg and opkg agree that trailing zeros should be ignored.

**Comment by jplitza** [comment]
@NeoRaider Then disregard my commit, but make sure to document that behavior somewhere - e.g. in the commit message, that currently says otherwise 😉. But of course also in the actual documentation.

**Comment by rubo77** [comment]
Is it also tested that 10>9?

for example update from 2018.2.0.9 to 2018.2.0.10

**Pull request merged**